src/integer_mod_q/z_q/distance.rs

```diff
@@ -98,6 +98,17 @@ impl Distance<&Zq> for Zq {
     }
 }
 
+impl Distance<Zq> for Zq {
+    type Output = Z;
+
+    // This reference links to the correct distance implementation
+    // just because the correct one is the first in this file.
+    /// Just calls [`Zq::distance<&Zq>`].
+    fn distance(&self, other: Zq) -> Self::Output {
```

Comment:
Missing doc comment (referrer would be sufficient)

Reply:
The last comment is already a doc comment.